modules/authentication/src/routes/middleware.ts

```diff
@@ -111,3 +112,53 @@ function getToken(headers: Headers, cookies: Cookies, reqType: 'access' | 'refre
   }
   return headerArgs[1] || tokenCookie;
 }
+
+export async function captchaMiddleware(call: ParsedRouterRequest) {
+  const config = ConfigController.getInstance().config;
+  const { acceptablePlatform, secretKey } = config.captcha;
+  const { clientId } = call.request.context;
+  if (!config.captcha.enabled) {
+    throw new GrpcError(status.INTERNAL, 'Captcha is disabled.');
+  }
+  const { captchaToken } = call.request.params;
+  if (isNil(captchaToken)) {
```

Comment:
You need to first check if captch SHOULD be validated for this platform before throwing an error

Reply:
The checks below also mess everything app. Fix the checks